pypeit/scripts/setup.py

```diff
@@ -82,5 +82,7 @@ def main(args):
                                     write_manual=args.manual_extraction,
                                     configs=[item.strip() for item in args.cfg_split.split(',')])
 
+        # Return -- allows this to be run by other scripts
+        return ps
```

Comment:
What other scripts need to run this?  I don't think we should be running `main` functions of a script anywhere except for the script.  If you want this functionality elsewhere, we should move it to a function within `PypeItSetup`.

Reply:
I have external code (and I think tests) that need access to the items
in `ps` after we have executed `main()`.  I could just copy the lines of `main()`
but am trying to avoid that duplicated code (and its maintenance).

Does this make sense?

Reply:
Yeah, I understand why, but I think it's bad practice.  I.e., script `main` functions shouldn't provide broad functionality, they should only use it.  I'm suggesting we move what you need from `main` into a new function, and then have both `main` and the other places where you need `ps` call that function.  That avoids duplicating code and restricts the use of `main`.